controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -204,6 +214,12 @@ public void initializeStreamWriters(final ClientFactory clientFactory,
                                 Throwable cause = ExceptionHelpers.getRealException(e);
                                 if (cause instanceof ScaleOperationExceptions.ScalePreConditionFailureException) {
                                     response.setStatus(ScaleResponse.ScaleStreamStatus.PRECONDITION_FAILED);
+                                } else if (cause instanceof ScaleOperationExceptions.ManualScaleTimedOut) {
+                                    // Note: timedout does not mean the scale operation failed. In fact scale operation once started,
```

Comment:
I am working on that itself. 
The good part is we dont expose manual scale apis via our client interfaces and since rest api for manual scale is also not available, so we can change the behaviour.. its only our tests that use the manual scale.
So i am making the change that @skrishnappa suggested. 
Here is what i am planning to provide -->
2 new apis on Controller interface and 1 new api on grpc apart from existing scaleStream (which will now return after successfully starting scale operation). 
Controller interface:: scaleStreamAwaitCompletion(executor), checkScaleStatus(epoch)
Grpc: checkScaleStatus(epoch)

ScaleStreamAwaitCompletion, executor) will internally call checkScaleStatus until scale status says 
scale complete.